python-spec/src/somacore/data.py

```diff
@@ -91,7 +91,7 @@ def shape(self) -> Tuple[int, ...]:
     @abc.abstractmethod
     def ndim(self) -> int:
         """The number of dimensions in this array."""
-        return len(self.shape)
+        raise NotImplementedError()
```

Comment:
This should still be `len(self.shape)`; there’s no need to make downstream users implement this.

Reply:
Did not see the earlier comment.

Reply:
Actually it looks like this still can be `len(self.shape)`. An implementation *can* override this, but it should always be valid.

Reply:
What would need changing there? It returns a tuple whose length will be `ndim`, so taking its `len` should be fine (as the spec implies). Am I missing something?